backend/open_webui/storage/provider.py

```diff
@@ -145,41 +145,41 @@ def delete_all_files(self) -> None:
 
 class GCSStorageProvider(StorageProvider):
     def __init__(self):
-        self.gcs_client = storage.Client.from_service_account_info(info=json.loads(GOOGLE_APPLICATION_CREDENTIALS_JSON))
-        self.bucket_name = self.gcs_client.bucket(GCS_BUCKET_NAME)
+        if GCS_PROJECT_ID:
+            self.gcs_client = storage.Client(project=GCS_PROJECT_ID)
+        if GOOGLE_APPLICATION_CREDENTIALS_JSON:
+            self.gcs_client = storage.Client.from_service_account_info(info=json.loads(GOOGLE_APPLICATION_CREDENTIALS_JSON))
+
+        self.bucket_name = GCS_BUCKET_NAME
```

Comment:
Normally property assignment go on the top of the init, and then if needed to be used in the init, is the property that gets used, not the function argument.